core/templates/dev/head/pages/exploration_editor/translation_tab/audio_translation_bar_directive.html

```diff
@@ -0,0 +1,227 @@
+          uib-tooltip="Re-take" tooltip-placement="bottom"
+          ng-if="!recorder.status.isRecording && audioBlob"
+          ng-click="reRecord()">
+    <i class="material-icons">&#xE028;</i>
+  </button>
+  <button class="btn oppia-audio-button"
+          uib-tooltip="Save" tooltip-placement="bottom"
+          ng-if="!recorder.status.isRecording && audioBlob"
+          ng-click="saveRecordedAudio()">
+    <i class="material-icons">&#xE161;</i>
+  </button>
+  <button class="btn oppia-audio-button"
+          uib-tooltip="Cancel" tooltip-placement="bottom"
+          ng-if="!recorder.status.isRecording && audioBlob"
+          ng-click="cancelRecording()">
+    <i class="material-icons">&#xE5C9;</i>
+  </button>
+</div>
+
+<div ng-if="isAudioAvailable && !audioIsUpdating">
+  <button class="btn oppia-audio-button"
+          ng-click="playPauseUploadedAudioTranslation()">
+    <i class="material-icons"><[isPlayingUploadedAudio() ? '&#xE035;' : '&#xE039;']></i>
+  </button>
+  <div class="oppia-slider-section">
+    <div ng-if="audioLoadingIndicatorIsShown">
+      <md-progress-linear md-mode="indeterminate"></md-progress-linear>
+    </div>
+    <div ng-if="!audioLoadingIndicatorIsShown">
+      <md-slider ng-model="track.progress" ng-model-options="{getterSetter: true}" aria-label="audio-slider">
+      </md-slider>
+    </div>
+  </div>
+  <div class="oppia-recording-timer" ng-if="!audioLoadingIndicatorIsShown">
+      <[getUploadedAudioTrimer()]>
+  </div>
+  <div class="oppia-audio-extra-buttons">
+    <button class="btn oppia-audio-button"
+            ng-disabled="!canTranslate"
+            uib-tooltip="Record new" tooltip-placement="bottom"
+            ng-click="updateAudio()">
+      <i class="material-icons">&#xE028;</i>
+    </button>
+    <button class="btn oppia-audio-button"
+            uib-tooltip="Delete" tooltip-placement="bottom"
+            ng-disabled="!canTranslate"
+            ng-click="openDeleteAudioTranslationModal()">
+      <i class="material-icons">&#xE872;</i>
+    </button>
+    <div class="btn dropdown oppia-audio-button" type="button"
+         ng-disabled="!canTranslate">
+      <div class="dropdown-toggle" type="button" data-toggle="dropdown">
+        <i class="material-icons">&#xE5D2;</i>
+      </div>
+      <ul class="dropdown-menu dropdown-menu-right">
+        <li ng-click="openAddAudioTranslationModal()"><a href="">Upload audio</a></li>
+        <li ng-if="!audioNeedsUpdate" ng-click="toggleAudioNeedsUpdate()"><a href="">Mark need update</a></li>
+        <li ng-if="audioNeedsUpdate" ng-click="toggleAudioNeedsUpdate()"><a href="">Unmark need update</a></li>
+      </ul>
+    </div>
+  </div>
+</div>
+
+<div class="oppia-translation-bottom-right-container" ng-if="showRecorderWarning">
+  <span>
+    <strong>Warning: </strong>Please keep this tab open until you save the recorded audio. If you navigate elsewhere before that, the recorded audio will be lost.
```

Comment:
So upon reading this, my instinct is to think Chrome tab, so that as long as I have the current browser tab open, the audio will not be last. I'm not sure if there's a way to make this more clear/explicit? Effectively, the user shouldn't do ANYTHING else while it's being recorded, converted, previewed, and saved. The main concern is conversion because that can take a long time for longer files, so the user should just sit there and not touch anything.